### PR TITLE
Improve successes for the `ocaml-ci` (for OCaml ≤ 4.12 + lower bounds)

### PR DESCRIPTION
**Pull request by smondet**

### CODE REVIEW

**Review by rr0gi — APPROVED** [review]

**Comment by smondet** [comment]
This brings the lower bound from 4.13 to 4.12. Next failure is on the `__FILENAME__` macro-ish value; added in 4.12 https://ocaml.org/manual/5.2/api/Stdlib.html .

**Comment by smondet** [comment]
In case anybody is also curious, double-checked that `__FUNCTION__` does not try to be smart about functors:

```ocaml
module F (P : sig val x : string end) = struct
  let p () = Printf.eprintf "x: %S, f: %S\n%!" P.x __FUNCTION__
end

module G = F(struct let x = "G-arg" end)
module H = F(struct let x = "H-arg" end)

let () =
  G.p ();
  H.p ();
  print_endline "Done."
```

```
 $ ocaml foo-bar.ml
x: "G-arg", f: "F.p"
x: "H-arg", f: "F.p"
Done.
```

**Comment by smondet** [comment]
The remaining failures are from `libevent` (missing `gcc` or `pkg-config` on *BSD and such).

File: web.ml [hunk 1]
```diff
@@ -301,7 +301,8 @@ module Http (IO : IO_TYPE) (Curl_IO : CURL with type 'a t = 'a IO.t) : HTTP with
         "url.full", `String url;
       ]
     in
-    let explicit_span = Possibly_otel.enter_manual_span ~__FUNCTION__ ~__FILE__ ~__LINE__ ~data:describe action_name in
+    let explicit_span =
+      Possibly_otel.enter_manual_span ~__FUNCTION__:"Web.Http.add_if_absent" ~__FILE__ ~__LINE__ ~data:describe action_name in
```
Comment (rr0gi) [inline web.ml:305]: why Web.Http.add_if_absent?

**Review by rr0gi — APPROVED** [review]

File: web.ml (hunk 1 above)
Comment (smondet) [reply web.ml:305]: That's what the value `__FUNCTION__` gives there right?

Comment (smondet) [reply web.ml:305]: Oops actually not, misread, sorry

Comment (smondet) [reply web.ml]: The _real_ value at that location seems to be `"Devkit_core__Web.Http.http_request'.(fun)"`, at least from `dune utop .`, not sure how it shows up in other build contexts.

**Review by rr0gi — APPROVED** [review]

**Pull request merged**